hyperspy/_signals/eds.py

```diff
@@ -617,3 +620,93 @@ def get_take_off_angle(self):
                                        elevation_angle)
 
         return TOA
+
+    def plot_xray_lines(self,
+                        xray_lines=None,
+                        only_lines=("a", "b"),
+                        only_one=False,
+                        **kwargs):
+        """
+        Annotate a spec.plot() with the name of the selected X-ray
+        lines
+
+        Parameters
+        ----------
+        xray_lines: {None, 'from_elements', list of string}
+            If None,
+            if `mapped.parameters.Sample.elements.xray_lines` contains a
+            list of lines use those.
+            If `mapped.parameters.Sample.elements.xray_lines` is undefined
+            or empty or if xray_lines equals 'from_elements' and
+            `mapped.parameters.Sample.elements` is defined,
+            use the same syntax as `add_line` to select a subset of lines
+            for the operation.
+            Alternatively, provide an iterable containing
+            a list of valid X-ray lines symbols.
+        only_lines : None or list of strings
+            If not None, use only the given lines (eg. ('a','Kb')).
+            If None, use all lines.
+        only_one : bool
+            If False, use all the lines of each element in the data spectral
+            range. If True use only the line at the highest energy
+            above an overvoltage of 2 (< beam energy / 2).
+        kwargs
+            The extra keyword arguments for plot()
+
+        See also
+        --------
+        set_elements, add_elements
+
+        """
+
+        if only_lines is not None:
+            only_lines = list(only_lines)
+            for only_line in only_lines:
+                if only_line == 'a':
+                    only_lines.extend(['Ka', 'La', 'Ma'])
+                elif only_line == 'b':
+                    only_lines.extend(['Kb', 'Lb1', 'Mb'])
```

Comment:
I would probably change `list(only_lines)` to `np.unique(only_lines)` to avoid having accidental duplicate entries.
Maybe also removing the general (e.g. "a" or "b") lines when found? 

I realise that in the end it does not matter at the moment, but it may some time later, if `_get_lines_from_elements` changes at some point